fe2-android/app/src/main/java/com/github/dedis/popstellar/utility/handler/data/LaoHandler.java

```diff
@@ -143,31 +156,38 @@ public static void handleStateLao(HandlerContext context, StateLao stateLao)
 
     // TODO: verify if lao/state_lao is consistent with the lao/update message
 
+    Lao lao = laoView.getLao();
+
     lao.setId(stateLao.getId());
     lao.setWitnesses(stateLao.getWitnesses());
     lao.setName(stateLao.getName());
     lao.setLastModified(stateLao.getLastModified());
     lao.setModificationId(stateLao.getModificationId());
 
     PublicKey publicKey = context.getKeyManager().getMainPublicKey();
-    if (lao.getOrganizer().equals(publicKey) || lao.getWitnesses().contains(publicKey)) {
-      context.getMessageSender().subscribe(lao.getChannel().subChannel("consensus")).subscribe();
+    if (laoView.isOrganizer(publicKey) || laoView.isWitness(publicKey)) {
+      context
+          .getMessageSender()
+          .subscribe(laoView.getChannel().subChannel("consensus"))
+          .subscribe();
```

Comment:
If the subscribtion fails, the app will crash

Reply:
unhelpful comment: that doesn't sound good 😅

Reply:
Can we catch that and log it ?

Reply:
Yep, good catch !